nexus/types/src/deployment.rs

```diff
@@ -1023,6 +1039,9 @@ impl From<crate::inventory::Dataset> for BlueprintDatasetConfigForDiff {
     fn from(dataset: crate::inventory::Dataset) -> Self {
         Self {
             name: dataset.name,
+            // TODO Should we know the dataset kind from inventory? We could
+            // probably infer it from the name, but yuck.
+            kind: None,
```

Comment:
Yeah, I didn't do this already, because I did not want inventory collection to fail if we see a dataset with a new "kind" value. So I can't actually map this directly to a `DatasetKind` value without having an `Unknown` variant, or without treating it as an optional.

Reply:
Would this not fall under the nexus -> sled-agent API compatibility?

Reply:
I suppose it could? I don't know if this is a rational concern, but the scope of inventory feels a little more like it could be arbitrary than other more-structured aspects of the API. For example, I wanted to ensure that "regardless of what shows up within hardware / datasets / etc, we can return something reasonable back to Nexus".

Basically, if it was possible for a "new, arbitrary dataset" to show up, I would still want that to be reportable through inventory, as much as possible -- even if Nexus can't **request** such a dataset, because it isn't a part of the strongly-typed Nexus -> Sled Agent API.